api/v2/apisixroute_types.go

```diff
@@ -111,7 +110,9 @@ type ApisixRouteHTTP struct {
 type ApisixRouteStream struct {
 	// Name is a unique identifier for the route. This field must not be empty.
 	Name string `json:"name" yaml:"name"`
-	// Protocol specifies the L4 protocol to match. Can be `tcp` or `udp`.
+	// Protocol specifies the L4 protocol to match. Can be `TCP` or `UDP`.
+	//
+	// +kubebuilder:validation:Enum=TCP;UDP
 	Protocol string `json:"protocol" yaml:"protocol"`
 	// Match defines the criteria used to match incoming TCP or UDP connections.
 	Match ApisixRouteStreamMatch `json:"match" yaml:"match"`
```

Comment:
The IngressPort in the match should add a jsonschema rule to verify the validity of the TCP port and ensure that the value is between 0 and 65535.